lib/stat_tracker.rb

```diff
@@ -14,19 +21,140 @@ def self.from_csv(data_files)
   end
 
   def game_collection
-    GameCollection.new
+    GameCollection.new(@games)
   end
 
   def team_collection
-    TeamCollection.new
+    TeamCollection.new(@teams)
   end
 
   def game_team_collection
-    GameTeamCollection.new
-  end 
+    GameTeamCollection.new(@game_teams)
+  end
 
   def highest_total_score
+    top_score = 0
+    game_collection.all.each do |game|
+      if game.away_goals.to_i + game.home_goals.to_i > top_score
+        top_score = game.away_goals.to_i + game.home_goals.to_i
+      end
+    end
+    top_score
+  end
+
+  def lowest_total_score
+    lowest_score = 1000000
+    game_collection.all.each do |game|
+      if game.away_goals.to_i + game.home_goals.to_i < lowest_score
+        lowest_score = game.away_goals.to_i + game.home_goals.to_i
+      end
+    end
+    lowest_score
+  end
+
+  def home_games
+    home_games = []
+    game_team_collection.all.flat_map do |game_team|
+      result = game_team.hoa == "home"
+        home_games << result
```

Comment:
Your use of flat map should negate the need to shovel in data into the empty home_games array. Rather, yo could set the whole iteration equal to home games.